### PR TITLE
Feature/uc 1223 eddsa support

### PR DESCRIPTION
**Pull request by Carlos-Augusto**
Adds support for EDDSA(ed25519)-based upp-verification

### CODE REVIEW

File: src/key.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+const getKey = (keyInBase64) => {
+  const pubKeyBuffer = Buffer.from(keyInBase64, 'base64');
+
+  let key;
+  if (pubKeyBuffer.length === 64) {
+    key = EC.getKey(keyInBase64);
+  } else if (pubKeyBuffer.length === 32) {
+    key = EDDSA.getKey(keyInBase64);
+  } else {
+    throw new Error('Invalid ECDSA Key Compressed');
```
Comment (leroxyl) [inline src/key.js]: throw new Error('Invalid ECDSA or EDDSA Key Compressed');

**Review by dagdelenmustafa — APPROVED** [review]
LGTM 👍

File: src/upp-parts.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+const upp = (uppInBase64) => {
+  const buff = Buffer.from(uppInBase64, 'base64');
+  const parts = decode(buff);
+
+  uppLengthCheck(parts);
+  uppVersionCheck(parts);
+
+  return {
+    bytes: buff,
+    decoded: decode(buff)
```
Comment (leroxyl) [inline src/upp-parts.js]: why not return "parts" here instead of decoding twice?

File: src/upp-parts.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  };
+};
+
+const getSignedAndSignature = (upp) => {
+  uppLengthCheck(upp.decoded);
+  uppVersionCheck(upp.decoded);
+
+  const signed = upp.bytes.subarray(0, upp.bytes.length - 66);
```
Comment (leroxyl) [inline src/upp-parts.js:74]: what happens if upp.bytes.length <= 66?

File: tests/upp.verify.test.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+'use strict';
+
+const assert = require('assert');
+const verify = require('../src/upp');
+
+describe('Verification', () => {
+  describe('upp', () => {
+    it('should get details from upp', () => {
+      const uppHash = 'liPEEM+T+e6L9EzBtxV79B2I5hbEQLKeDjxuX6RTp6/kKnsJR+cd3exsAqA/8oJXdYjzVvfWG3I3QXeqzTdAgJj8No6sL0ltaSGWjzEwBAy+fx+ZdCkAxCB8OqIKiOGsBRJoM8hSLnaawKFfIHdVFYVYBP5XRaEWJMRAPYfV3BJ4goY6HUxSNcB6Wu48Y+5iRqsuRdUT4dlidzaD9bjub7DxN75sXzf5uOgn26lZ1asuPsfKPWaYuciXTQ==';
```
Comment (leroxyl) [inline tests/upp.verify.test.js]: uppHash is not a UPP hash, maybe it should be called uppBase64

**Review by leroxyl — COMMENTED** [review]
looks good, just a few minor comments.

File: src/key.js (hunk 1 above)
Comment (Carlos-Augusto) [reply src/key.js]: Ah, of course, I need to adjust the message, thank you.

File: src/upp-parts.js (hunk 2 above)
Comment (Carlos-Augusto) [reply src/upp-parts.js]: This is an error. Good catch. I will fix it.

File: src/upp-parts.js (hunk 3 above)
Comment (Carlos-Augusto) [reply src/upp-parts.js:74]: Very interesting, see this quick experiment:
image
The resultant buffer eventually gets empty.

Comment (Carlos-Augusto) [reply src/upp-parts.js:74]: I will add a error though.

File: tests/upp.verify.test.js (hunk 4 above)
Comment (Carlos-Augusto) [reply tests/upp.verify.test.js]: Yes, you are right.

File: src/key.js (hunk 1 above)
Comment (Carlos-Augusto) [reply src/key.js]: Done

**Review by leroxyl — APPROVED** [review]

**Pull request merged**